dll/win32/browseui/shellbrowser.cpp

```diff
@@ -783,11 +786,12 @@ HRESULT CShellBrowser::Initialize()
 
     fToolbarProxy.Initialize(m_hWnd, clientBar);
 
-    LoadSettings();
+    LoadCabinetState();
+    ::SendMessageW(fClientBars[BIInternetToolbar].hwnd, BWM_SETTINGCHANGE, 0, (LPARAM)&m_settings);
```

Comment:
An alternative could be for the toolbar itself to query its parent window for the settings structure (via a window message) :thinking: (when e.g. it gets "attached" to it via the SetSite call or sthg....)

Reply:
That could be a better solution, I'll look into it. Thanks!

Reply:
I've got something like this up and running, I will clean it up real quick.